src/Evolution/Systems/GeneralizedHarmonic/CharFields.hpp

```diff
@@ -0,0 +1,89 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+namespace Tags {
+template <typename Tag>
+struct Normalized;
+}  // namespace Tags
+/// \endcond
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace GeneralizedHarmonic {
+/*!
+ * \ingroup GeneralizedHarmonic
+ * \brief Compute the characteristic fields for the generalized harmonic system.
+ *
+ * Computes the fields as described in "A New Generalized Harmonic
+ * Evolution System" by Lindblom et. al, https://arxiv.org/abs/gr-qc/0512093.
+ * The characteristic fields' names used here differ from this paper:
+ *
+ * \f{align*}
+ * \mathrm{SpECTRE} && \mathrm{Lindblom} \\
+ * u^{\psi}_{ab} && u^\hat{0}_{ab} \\
+ * u^{\pm}_{ab} && u^{\hat{1}\pm}_{ab} \\
+ * u^0_{iab} && u^\hat{2}_{iab}
+ * \f}
+ *
+ * The fields \f$v\f$, are given by:
+ * \f{align*}
+ * u^{\psi}_{ab} =& \psi_{ab} \\
+ * u^{\pm}_{ab} =& \Pi_{ab} \pm n^i \Phi_{iab} - \gamma_2\psi_{ab} \\
+ * u^0_{iab} =& (\delta^k_i - n_i n^k) \Phi_{kab} := P^k_i \Phi_{kab}
+ * \f}
+ *
+ * where \f$\psi_{ab}\f$ is the spacetime metric, \f$\Pi_{ab}\f$ and
+ * \f$\Phi_{iab}\f$ are evolved GH fields introduced by first derivatives of
+ * \f$\psi_{ab}\f$; \f$\gamma_2\f$ is a constraint damping parameter, and
+ * \f$n_k\f$ is the unit normal to the surface.
+ */
```

Comment:
Please cite equation numbers from Lindblom+ (2005) for each characteristic field's definition for each characteristic field. Please define GH, or better yet, just spell out generalized harmonic.

Reply:
done